plugin/src/sbt-test/scalafmt-sbt/sbt/test

```diff
@@ -54,3 +54,17 @@ $ newer p8/src/main/scala/Test2.scala timestamp
 -> p12/scalafmtCheck
 
 > check
+
+> p13/scalafmt
+# prevent reading the source without changing the mtime to detect actual/uncached invocation of scalafmt
+$ exec chmod 000 p13/src/main/scala/Test.scala
```

Comment:
not particularly proud of this and it's somewhat fragile (in case the caching strategy changes to `FileInfo.hash`, this will bring a false positive), but instrumenting `ScalafmtPlugin.globalInstance` sounded like a lot of work for this, and spying on the output log of the task felt even more fragile.